sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -666,12 +665,26 @@ class Analyzer(
   }
 
   /**
-   * Resolve table relations with concrete relations from v2 catalog.
+   * Resolve relations to temp views. This is not an actual rule, and is only called by
+   * [[ResolveTables]].
+   */
+  object ResolveTempViews extends Rule[LogicalPlan] {
+    def apply(plan: LogicalPlan): LogicalPlan = plan.resolveOperatorsUp {
+      case u @ UnresolvedRelation(Seq(part1)) =>
+        v1SessionCatalog.lookupTempView(part1).getOrElse(u)
+      case u @ UnresolvedRelation(Seq(part1, part2)) =>
```

Comment:
This needs to check whether `part1` is a known catalog. If it is a catalog, then it isn't a temp view reference because catalog resolution happens first.

Not needing to remember that is the purpose of the extractors. I think it would be better to continue using the extractor:

```scala
  case u @ UnresolvedRelation(AsTemporaryTableIdentifier(ident)) =>
    ident.database match {
      case Some(db) =>
        v1SessionCatalog.lookupGlobalTempView(db, ident.table).getOrElse(u)
      case None =>
        v1SessionCatalog.lookupTempView(ident.table).getOrElse(u)
    }
```

Reply:
as discussed in the sync, we decide to treat the global temp view name prefix `global_temp` as a special catalog, so that it won't be masked by user-supplied catalog.

Reply:
If this is intended to only match `global_temp` then it should match `Seq(GLOBAL_TEMP_NAMESPACE, name)` instead of matching `part1`.

Reply:
unfortunately, `global_temp` is not a constant, it's a static SQL conf that users can set before starting a Spark application(not at runtime)

Reply:
Yeah, that's why I went ahead with the merge. I think the code is currently correct.

Still, it would be nice to use the runtime setting here in the matcher instead.